Bound just size variables in bound_sympy

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #105139
* #105138
* __->__ #105155

We also bound them as starting on 2, because of 0,1 specialisation.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

lezcano committed:
Bound just size variables in bound_sympy

We also bound them as starting on 2, because of 0,1 specialisation.

[ghstack-poisoned]

lezcano committed:
Update on "Bound just size variables in bound_sympy"


We also bound them as starting on 2, because of 0,1 specialisation.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

ezyang commented:
We were discussing this on Slack and the claim I made was you can't actually use the s prefix to test if something was 0/1 specialized.

TBH, I would rather we fix it differently: can we arrange the invariant that all allocated symbols always have a ValueRange associated with it on construction? This just involves modifying some inductor code I think.

lezcano committed:
Update on "Bound just size variables in bound_sympy"


We also bound them as starting on 2, because of 0,1 specialisation.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

lezcano commented:
OK, given that this function is of general purpose, we are a bit more careful  on how to bound the symbols, and we just use the info they were created with to do so.

I left a comment as, in general, I agree it'd be nicer to always attach a `ValueRange` to a given symbol when instantiated.

lezcano committed:
Update on "Bound just size variables in bound_sympy"


We also bound them as starting on 2, because of 0,1 specialisation.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

ezyang reviewed (approved):
nice
